lib/ocpp/v201/charge_point.cpp

```diff
@@ -3207,6 +3232,97 @@ void ChargePoint::handle_set_charging_profile_req(Call<SetChargingProfileRequest
     this->send<SetChargingProfileResponse>(call_result);
 }
 
+void ChargePoint::handle_clear_charging_profile_req(Call<ClearChargingProfileRequest> call) {
+    EVLOG_debug << "Received ClearChargingProfileRequest: " << call.msg << "\nwith messageId: " << call.uniqueId;
+    const auto msg = call.msg;
+    ClearChargingProfileResponse response;
+    response.status = ClearChargingProfileStatusEnum::Unknown;
+
+    // K10.FR.06
+    if (msg.chargingProfileCriteria.has_value() and
+        msg.chargingProfileCriteria.value().chargingProfilePurpose.has_value() and
+        msg.chargingProfileCriteria.value().chargingProfilePurpose.value() ==
+            ChargingProfilePurposeEnum::ChargingStationExternalConstraints) {
+        response.statusInfo = StatusInfo();
+        response.statusInfo->reasonCode = "InvalidValue";
+        response.statusInfo->additionalInfo = "ChargingStationExternalConstraintsInClearChargingProfileRequest";
```

Comment:
I think we can make this string a bit more readable with spaces instead of camelcase. There are no rules for this string right?

Reply:
So far this is based on the behavior of `handle_set_charging_profile_req`.

Reply:
Some thoughts:

* This is modeled after `ProfileValidationResultEnum` I assume. Looking at this and some upcoming code for GetCompositeSchedule, I wonder if we should rename that to be something like `SmartChargingResultEnum` (or have a separate, similar enum)
* Related, the conversion we use essentially just uses the name of the enum. If we want these to be more readable across the board, we should probably change those too.

I think it would be good to standardize this somewhat, and then in cases like this we could generate `statusInfo` from an enum.

Reply:
I like this idea, probably its better to not rename but have a separate enum for that, since `ProfileValidationResultEnum` is really a result of the validation of the profile. I would not change this as part of this PR (and dont expect e.g. https://github.com/EVerest/libocpp/pull/745 to something for it) but rather add this as part of a subsequent PR.